jira/jira.go

```diff
@@ -1,41 +1,101 @@
 package jira
 
 import (
+	gojira "github.com/andygrunwald/go-jira"
 	"github.com/go-chat-bot/bot"
 	"os"
 	"regexp"
 	"strings"
 )
 
 const (
-	pattern = "(^|\\s)+[A-z]{3}-[0-9]+\\b"
-	env     = "JIRA_ISSUES_URL"
+	pattern    = ".*?([A-Z]+)-([0-9]+)\\b"
+	env        = "JIRA_ISSUES_URL"
+	userEnv    = "JIRA_USER"
+	passEnv    = "JIRA_PASS"
+	baseURLEnv = "JIRA_BASE_URL"
 )
 
 var (
-	url string
-	re  = regexp.MustCompile(pattern)
+	url      string
+	baseURL  string
+	jiraUser string
+	jiraPass string
+	projects map[string]gojira.Project
+	client   *gojira.Client
+	re       = regexp.MustCompile(pattern)
 )
 
-func getIssue(text string) string {
-	if re.MatchString(text) {
-		return re.FindString(text)
+func getProjects() map[string]gojira.Project {
+	req, _ := client.NewRequest("GET", "rest/api/2/project", nil)
+
+	projectObjects := new([]gojira.Project)
+	_, err := client.Do(req, projectObjects)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
KInda similar to comment above, since this is called during startup I though panic might be better than letting people debug why the bot is not responding to any JIRA mentions. Logging the error should help with that though so I'll tweak